### ISSUE TITLE
Street Fighter® IV (21660)

### ISSUE THREAD

**Issue by guihkx** [ValveSoftware/Proton#4074]
# Compatibility Report
- Name of the game with compatibility issues: Street Fighter® IV
- Steam AppID of the game: 21660

## System Information
- GPU: GTX 660
- Driver/LLVM version: NVIDIA 450.57
- Kernel version: 5.7.8
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/guihkx/0d8e636ffbf43f36170102c436eda896/raw/e7968f1ef3c0bb036d0614d92e97e1377fb61e71/gistfile1.txt
- Proton version: 5.0-9

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-21660.log](https://gist.githubusercontent.com/guihkx/a1e872c8f332aef84d8ee8d345e19681/raw/a2362bef98b264a6a80a5be594f393fcfc9b5330/steam-21660.log)

## Symptoms 
Once you click to "Launch STREET FIGHTER IV" in the game's launcher, an empty black window opens and the game seems to freeze.

## Reproduction
Just try launching the game.

**Comment by SergioCW** [comment]
Works for me with `PROTON_NO_ESYNC=1`

**Comment by guihkx** [comment]
Yes, of course! I totally forgot to try the game with `PROTON_NO_ESYNC=1`! Additionally, I also had to set `PROTON_NO_FSYNC=1`, because I'm using a custom kernel with FSYNC support.

Other than that, it runs pretty amazing. I get a constant 120 fps with no stuttering whatsoever, with the graphics maxed out. Here are the benchmark results:

![20200720205459_1](https://user-images.githubusercontent.com/626206/87997460-86629180-cacb-11ea-9153-79c6ab6a5773.jpg)

Works beautifully with my DS4 gamepad too! But that's thanks to Steam, I think.

The only thing it doesn't work is online multiplayer, but that's expected since this game has GFWL integration and that's a defunct service.

**Comment by zfigura** [comment]
It uses PulseEvent(); that's unfixable with esync.

**Comment by Managor** [comment]
So the game now works out of the box, I see.

**Comment by guihkx** [comment]
> So the game now works out of the box, I see.

Yup, it does. Proton now automatically disables esync/fsync for this title. I wonder if these game-specific workarounds are private because I couldn't find them in this repo...

**Comment by kisak-valve** [comment]
The runtime config options are being applied by Steam's whitelist config, even though this particular game is not currently whitelisted.

**Comment by guihkx** [comment]
So this list of workarounds is indeed private? Because I'll be opening a new issue for a different game that also uses PulseEvent() and it'd be awesome if Steam automatically disabled esync/fsync for that one too.
